test/integration/controllermanager/certificatesigningrequest/certificatesigningrequest_test.go

```diff
@@ -69,31 +71,33 @@ var _ = Describe("CSR autoapprove controller tests", func() {
 		})
 	})
 
-	Context("non shoot client certificate", func() {
+	Context("non seed client certificate", func() {
 		BeforeEach(func() {
 			certificateSubject = &pkix.Name{
 				CommonName: "csr-autoapprove-test",
 			}
-			csrData, err := certutil.MakeCSR(privateKey, certificateSubject, nil, nil)
+			csrData, err = certutil.MakeCSR(privateKey, certificateSubject, nil, nil)
 			Expect(err).NotTo(HaveOccurred())
 			csr.Spec.Request = csrData
 		})
+
 		It("should ignore the CSR and do nothing", func() {
 			Eventually(logBuffer).Should(gbytes.Say("Ignoring CSR, as it does not match the requirements for a seed client"))
-			Expect(csr.Status.Conditions).To(BeEmpty())
+			Consistently(csr.Status.Conditions).Should(BeEmpty())
```

Comment:
Async assertions only make sense if the actual value changes.
You would need to pass a function that retrieves the CSR from the client.

Reply:
This is still not resolved. You have to move the `Get` call into the `Consistently` function.